api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -113,4 +113,18 @@ message LocalRateLimit {
   // Disabled by default.
   common.ratelimit.v3.XRateLimitHeadersRFCVersion enable_x_ratelimit_headers = 12
       [(validate.rules).enum = {defined_only: true}];
+
+  enum VhRateLimitsOptions {
+    // Use the virtual host rate limits unless the route has a rate limit policy.
+    OVERRIDE = 0;
+
+    // Use the virtual host rate limits even if the route has a rate limit policy.
+    INCLUDE = 1;
+
+    // Ignore the virtual host rate limits even if the route does not have a rate limit policy.
+    IGNORE = 2;
+  }
+
+  // Specifies if the local rate limit filter should include the virtual host rate limits.
+  VhRateLimitsOptions vh_rate_limits = 13 [(validate.rules).enum = {defined_only: true}];
```

Comment:
IMHO the enum from `envoy.extensions.filters.http.ratelimit.v3` should be used.

It would have been better to move this enum to some common location, but unfortunately I think it will break the current API.

Reply:
yes, thanks, i will use `envoy.extensions.filters.http.ratelimit.v3` instead.

Reply:
`x1b[0m/build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy_api/envoy/extensions/filters/http/local_ratelimit/v3/BUILD:7:18: GoCompilePkg external/envoy_api/envoy/extensions/filters/http/local_ratelimit/v3/pkg_go_proto.a failed: (Exit 1): builder failed: error executing command (from target @envoy_api//envoy/extensions/filters/http/local_ratelimit/v3:pkg_go_proto) bazel-out/k8-opt-exec-2B5CBBC6/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix linux_amd64 -src ... (remaining 85 arguments skipped)\n\nUse --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging\nbazel-out/k8-fastbuild/bin/external/envoy_api/envoy/extensions/filters/http/local_ratelimit/v3/pkg_go_proto_/github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.pb.validate.go:43:6: undefined: "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ratelimit/v3".XRateLimitHeadersRFCVersion\nbazel-out/k8-fastbuild/bin/external/envoy_api/envoy/extensions/filters/http/local_ratelimit/v3/pkg_go_proto_/github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.pb.validate.go:343:11: assignment mismatch: 2 variables but 1 value\nbazel-out/k8-fastbuild/bin/external/envoy_api/envoy/extensions/filters/http/local_ratelimit/v3/pkg_go_proto_/github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.pb.validate.go:343:14: undefined: "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3".RateLimitPerRoute_VhRateLimitsOptions_name\ncompilepkg: error running subcommand external/go_sdk/pkg/tool/linux_amd64/compile: exit status 2\n\x1b[32mINFO: \x1b[0mElapsed time: 189.979s, Critical Path: 20.43s\n\x1b[32mINFO: \x1b[0m1278 processes: 24 internal, 1254 processwrapper-sandbox.\n\x1b[31m\x1b[1mFAILED:\x1b[0m Build did NOT complete successfully\n\x1b[31m\x1b[1mFAILED:\x1b[0m Build did NOT complete successfully\n\x1b[0m'`

@adisuissa i tried use `envoy.extensions.filters.http.ratelimit.v3`, but the `bazel.api` check failed, detailed errors are above. it seems that the go_package name of `envoy.extensions.common.ratelimit.v3` and `envoy.extensions.filters.http.ratelimit.v3` conflict?

Reply:
Yeah, sorry about that.
I guess the way to get around this is to do something similar to what was done for [`XRateLimitHeadersRFCVersion `](https://github.com/envoyproxy/envoy/blob/370f60b81d936a106a09689d355d45fd0da7d6b5/api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto#L36) (context in PR #18157):
1. Copy the enum into common.
2. In the local ratelimit use the enum from common.
3. Add comments for the definition and usage of the enum (similar to [this](https://github.com/envoyproxy/envoy/blob/370f60b81d936a106a09689d355d45fd0da7d6b5/api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto#L35) and [this](https://github.com/envoyproxy/envoy/blob/370f60b81d936a106a09689d355d45fd0da7d6b5/api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto#L108)).

Reply:
ok, thanks！